src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedVariantContextUtils.java

```diff
@@ -523,4 +523,217 @@ private static int getMinBaseQualityForAlleleInRead(final PileupElement pileupEl
+        // length evenly divisible by 3 because that's how many bases are read at once:
+        return ((Math.abs( reference.length() - alternate.length() ) % 3) != 0);
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute a frameshift mutation.
+     * This does not take into account the cases where either or both of the alleles overlap splice sites.
+     * @param startPos Genomic start position (1-based, inclusive) of the variant.  Must be > 0.
+     * @param refEnd Genomic end position (1-based, inclusive) of the reference allele.  Must be > 0.
+     * @param altEnd Genomic end position (1-based, inclusive) of the alternate allele.  Must be > 0.
+     * @return {@code true} if replacing the reference with the alternate results in a frameshift.  {@code false} otherwise.
+     */
+    public static boolean isFrameshift(final int startPos, final int refEnd, final int altEnd) {
+
+        ParamUtils.isPositive( startPos, "Genomic positions must be > 0." );
+        ParamUtils.isPositive( refEnd, "Genomic positions must be > 0." );
+        ParamUtils.isPositive( altEnd, "Genomic positions must be > 0." );
+
+        final int refLength = refEnd - startPos + 1;
+        final int altLength = altEnd - startPos + 1;
+
+        // We know it's a frameshift if we have a replacement that is not of a
+        // length evenly divisible by 3 because that's how many bases are read at once:
+        return ((Math.abs( refLength - altLength ) % 3) != 0);
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute an insertion mutation.
+     * @param reference The reference {@link Allele}.   Must not be {@code null}.
+     * @param alternate The alternate / variant {@link Allele}.   Must not be {@code null}.
+     * @return {@code true} if replacing the reference with the alternate results in an insertion.  {@code false} otherwise.
+     */
+    public static boolean isInsertion(final Allele reference, final Allele alternate) {
+
+        Utils.nonNull(reference);
+        Utils.nonNull(alternate);
+
+        // If we have more bases in the alternate, we have an insertion:
+        return reference.length() < alternate.length();
+    }
+
+    /**
+     * Determines whether the given reference and alternate alleles constitute an insertion mutation.
+     * @param reference A {@link String} containing the bases of the reference allele.  Must not be {@code null}.
+     * @param alternate A {@link String} containing the bases of the alternate / variant allele.  Must not be {@code null}.
+     * @return {@code true} if replacing the reference with the alternate results in an insertion.  {@code false} otherwise.
+     */
+    public static boolean isInsertion(final String reference, final String alternate) {
+
+        Utils.nonNull(reference);
+        Utils.nonNull(alternate);
+
+        if ( reference.contains("*") ) {
```

Comment:
But when an Allele instance is reference and `toString()` is called, the star is appended.  
I think you should support that case.

Reply:
I expected that the user would create the allele strings using `new String(allele.getBases())`.

I'll make this more permissive.